Neos.Flow/Classes/ResourceManagement/ResourceManager.php

```diff
@@ -134,6 +136,20 @@ protected function initialize()
         $this->initialized = true;
     }
 
+    protected function persistResource(PersistentResource $resource, CollectionInterface $collection, string $sourceHint = ''): void
+    {
+        $this->resourceRepository->add($resource);
+        GenericObjectValidator::onValidated($resource, function (Result $result) use ($resource) {
+            if ($result->hasErrors()) {
+                $this->deleteResource($resource, false);
```

Comment:
Just a `resourceRepository->remove($resource)` does not suffice, because the file has already been stored on the storage, which should not have happened in the first place...